static-assets/components/cstudio-forms/data-sources/embedded-content.js

```diff
@@ -66,7 +68,16 @@ YAHOO.extend(CStudioForms.Datasources.EmbeddedContent, CStudioForms.CStudioFormD
         false,
         {
           success: function (contentTO, editorId, name, value) {
-            control.insertItem(name, value, null, null, _self.id);
+            if (!_self.inserted) {
+              control.insertItem(name, value, null, null, _self.id);
+              _self.inserted = true;
+            } else {
+              // Recently added item is in the last position
+              const itemIndex = control.items.length - 1;
+              // When adding a new embedded content, the form may be saved using the 'Save draft' or the 'Save & Minimize'
+              // options. When it happens, the next save operation will be an edition, not a creation.
+              control.updateEditedItem({ value }, _self.id, itemIndex);
+            }
```

Comment:
- What about the path above (the if case)?
- I'm not completely in context, but not sure the last position will always contain the item that's being created or edited?

Reply:
In this case (when creating a new node-selector content) it'll always be at the last position. Also, the update call in here will refer only to the newly created item, that again, will be at the last position